code/modules/shuttle/on_move.dm

```diff
@@ -100,6 +101,11 @@ All ShuttleMove procs go here
 	if(rotation)
 		shuttleRotate(rotation)
 
+	if (src.type != /atom/movable/lighting_object)
+		var/turf/newT = get_turf(src)
+		if (newT.z != oldT.z)
+			onTransitZ(oldT.z, newT.z)
```

Comment:
I really don't like this solution and would prefer we find something closer to the source of the problem

Reply:
the problem as i understand it is that afterShuttleMove does lighting updates, so some of the moved_atoms will be qdel'd and some might actually become null references depending on how check_tick plays out. Another way to get this working is to check for QDELETED after update_lighting and return, yet another is to never move lighting objects.